pkg/collector/corechecks/snmp/snmp.go

```diff
@@ -38,6 +38,7 @@ type Check struct {
 	discovery                  *discovery.Discovery
 	sessionFactory             session.Factory
 	workerRunDeviceCheckErrors *atomic.Uint64
+	interfaceBandwidthState    *report.InterfaceBandwidthState
```

Comment:
Shall we store `report.InterfaceBandwidthState` state on `DeviceCheck` (devicecheck.go) instead of `Check` (snmp.go) ?

The main reason is to make the scope of the state more localised to where it's needed.

That will also help avoid having to pass the DeviceID down (`deviceCk.GetDeviceID()`).